src/main/java/de/fearnixx/jeak/service/token/TokenConfiguration.java

```diff
@@ -75,14 +75,32 @@ public TokenScope getTokenScopes(String token) {
         return new TokenScope(tokenScopes);
     }
 
+    /**
+     * Save the provided token for the provided scope.
+     *
+     * @param token The token as {@link String} to save.
+     * @param tokenScope The scope as {@link TokenScope} the token is valid for.
+     */
     public void saveToken(String token, TokenScope tokenScope) {
         ConfigNode child = new ConfigNode();
         child.setList();
         child.appendValue(tokenScope.getScopeSet());
         getConfig().put(token, child);
     }
 
+    /**
+     * Delete the provided token. Deleting a not existing token is considered a successful removal,
+     * since the token isn't existing after the {@link TokenConfiguration#deleteToken} method call.
+     *
+     * @param token The token as {@link String} to be deleted.
+     * @return true if the token was successfully deleted or doesn't exist,
+     * false otherwise
+     */
     public boolean deleteToken(String token) {
-        return getConfig().remove(token) != null;
+        boolean deleteSuccessful = false;
+        if (getConfig().getNode(token) != null && !getConfig().getNode(token).isVirtual()) {
```

Comment:
``#getNode`` will _never_ return ``null``. It is safe to just check ``#isVirtual``.  
This code will also return ``false`` for tokens that don't exist. ``deleteSuccessful`` should default to ``true`` with the current checks in place.

Reply:
I think this is fixed now